$NetBSD: patch-ab,v 1.46 2024/12/29 10:23:26 wiz Exp $

CC_VERSION_OPTS for gcc is modified to avoid going through the link
stage. If pkgsrc appends or prepends any linker flags through
CWRAPPERS (e.g. for MKRELRO), getting the version of the compiler may
fail because GCC will try to link the nonexistent program.

As of libheif-1.18.0, builtin_h265_decoder and so on are not available
anymore.

Fix gdkpixbufloader name for svg.

--- configure.orig	2024-05-03 00:36:33.000000000 +0000
+++ configure
@@ -21054,11 +21054,11 @@ WARNING: GCC 7.2.0 has a serious bug aff
          See https://bugzilla.gnome.org/show_bug.cgi?id=787222"
   fi
   # For GCC, use -v which has more information than --version.
-  CC_VERSION_OPTS="-v"
+  CC_VERSION_OPTS="-c -v"
 else
   # This is the list of common options used by autotools to check
   # versions for various compilers.
-  CC_VERSION_OPTS="--version -v -V -qversion"
+  CC_VERSION_OPTS="-c --version -v -V -qversion"
 fi
 
 for CC_VERSION_OPT in $CC_VERSION_OPTS; do
@@ -30417,13 +30417,13 @@ can_export_heic=no
 can_import_avif=no
 can_export_avif=no
 if test "x$have_libheif" = xyes; then
-  can_import_heic=`$PKG_CONFIG --variable=builtin_h265_decoder libheif`
-  can_export_heic=`$PKG_CONFIG --variable=builtin_h265_encoder libheif`
+  can_import_heic=yes
+  can_export_heic=yes
   if test "x$can_import_heic" = xyes; then
     MIME_TYPES="$MIME_TYPES;image/heif;image/heic"
   fi
-  can_import_avif=`$PKG_CONFIG --variable=builtin_avif_decoder libheif`
-  can_export_avif=`$PKG_CONFIG --variable=builtin_avif_encoder libheif`
+  can_import_avif=yes
+  can_export_avif=yes
   if test "x$can_import_avif" = xyes; then
     MIME_TYPES="$MIME_TYPES;image/avif"
   fi
@@ -33906,7 +33906,7 @@ fi
     # So let's test files the shell way.
     if (test "x$platform_win32" = "xyes" &&
         test -f "$gdk_pixbuf_moduledir/libpixbufloader-svg.dll") ||
-       test -f "$gdk_pixbuf_moduledir/libpixbufloader-svg.so"; then
+       test -f "$gdk_pixbuf_moduledir/libpixbufloader_svg.so"; then
       # We must not use $PKG_CONFIG nor PKG_CHECK_* macros because we need
       # to make sure we use the native pkg-config (in case we cross-compile).
       if pkg-config --atleast-version=2.56.2 glib-2.0 &&
@@ -34799,7 +34799,7 @@ esac
 
 gimpdatadir="$datadir/$PACKAGE/2.0"
 gimpplugindir="$libdir/$PACKAGE/2.0"
-gimpsysconfdir="$sysconfdir/$PACKAGE/2.0"
+gimpsysconfdir="$sysconfdir"
 
 
 
